extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampExtractorHelper.java

```diff
@@ -0,0 +1,161 @@
+import org.schabi.newpipe.extractor.localization.DateWrapper;
+import org.schabi.newpipe.extractor.utils.Utils;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Locale;
+
+public class BandcampExtractorHelper {
+
+    /**
+     * <p>Get an attribute of a web page as JSON
+     *
+     * <p>Originally a part of bandcampDirect.</p>
+     *
+     * @param html     The HTML where the JSON we're looking for is stored inside a
+     *                 variable inside some JavaScript block
+     * @param variable Name of the variable
+     * @return The JsonObject stored in the variable with this name
+     */
+    public static JsonObject getJsonData(final String html, final String variable)
+            throws JsonParserException, ArrayIndexOutOfBoundsException {
+        final Document document = Jsoup.parse(html);
+        final String json = document.getElementsByAttribute(variable).attr(variable);
+        return JsonParser.object().from(json);
+    }
+
+    /**
+     * Translate all these parameters together to the URL of the corresponding album or track
+     * using the mobile api
+     */
+    public static String getStreamUrlFromIds(final long bandId, final long itemId, final String itemType)
+            throws ParsingException {
+
+        try {
+            final String jsonString = NewPipe.getDownloader().get(
+                    "https://bandcamp.com/api/mobile/22/tralbum_details?band_id=" + bandId
+                            + "&tralbum_id=" + itemId + "&tralbum_type=" + itemType.substring(0, 1))
+                    .responseBody();
+
+            return JsonParser.object().from(jsonString).getString("bandcamp_url").replace("http://", "https://");
+
+        } catch (final JsonParserException | ReCaptchaException | IOException e) {
+            throw new ParsingException("Ids could not be translated to URL", e);
+        }
+
+    }
+
+    /**
+     * Concatenate all non-null and non-empty strings together while separating them using
+     * the comma parameter
+     */
+    public static String smartConcatenate(final String[] strings, final String comma) {
+        final StringBuilder result = new StringBuilder();
+
+        // Remove empty strings
+        final ArrayList<String> list = new ArrayList<>(Arrays.asList(strings));
+        for (int i = list.size() - 1; i >= 0; i--) {
+            if (Utils.isNullOrEmpty(list.get(i)) || list.get(i).equals("null")) {
+                list.remove(i);
+            }
+        }
+
+        // Append remaining strings to result
+        for (int i = 0; i < list.size(); i++) {
+            result.append(list.get(i));
+
+            if (i != list.size() - 1) {
+                // This is not the last iteration yet
+                result.append(comma);
+            }
+
+        }
+
+        return String.valueOf(result);
+    }
```

Comment:
Same here, it should go to `Utils`. Maybe you can even adjust the `join` method there

Reply:
Moved and fixed this method